src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs

```diff
@@ -2374,7 +2381,7 @@ public static void Reverse(ref byte buf, nuint length)
                 buf = ref Unsafe.Add(ref buf, numIters * numElements);
                 length -= numIters * numElements * 2;
             }
-            else if (Ssse3.IsSupported && (nuint)Vector128<byte>.Count * 2 <= length)
+            else if ((Ssse3.IsSupported || WasmBase.IsSupported) && (nuint)Vector128<byte>.Count * 2 <= length)
```

Comment:
Same here, but using `Vector128.Shuffle(tempFirst, Vector128.Create((byte)15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0));` and `Vector128.Shuffle(tempLast, Vector128.Create((byte)15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0));`

The `Create` calls do need to be "inline" due to a JIT limitation in .NET 7, but they will get optimized and hoisted regardless since we have the relevant `GT_CNS_VEC` support in RyuJIT (Mono LLVM has something similar).